src/sections/crosshair.js

```diff
@@ -0,0 +1,283 @@
+				"inDirectionSnap must be of type number",
+			);
+		this._direction = inDirection;
+		return this;
+	}
+
+	directionSnap(inDirectionSnap) {
+		if (typeof inDirectionSnap !== "number")
+			throw this.sequence._customError(
+				this,
+				"directionSnap",
+				"inDirectionSnap must be of type number",
+			);
+		this._config["snap"]["direction"] = inDirectionSnap;
+		return this;
+	}
+
+	lockManualRotation(inBool = true){
+		if (typeof inBool !== "boolean")
+			throw this.sequence._customError(
+				this,
+				"preventUserRotation",
+				"inBool must be of type number",
+			);
+		this._config["lockManualRotation"] = inBool;
+		return this;
+	}
+
+	borderColor(inColor) {
+		this._borderColor = lib.parseColor(inColor);
+		return this;
+	}
+
+	fillColor(inColor) {
+		this._fillColor = lib.parseColor(inColor);
+		return this;
+	}
+
+	lockLocation(inLocation, inOptions = {}) {
+		inOptions = foundry.utils.mergeObject(
+			{
+				offsetDistance: 0,
+				minDistance: null,
+				maxDistance: null,
+				edge: false
+			},
+			inOptions,
+		);
+		if (typeof inOptions.offsetDistance !== "number")
+			throw this.sequence._customError(
+				this,
+				"lockLocation",
+				"inOptions.offsetDistance must be of type number",
+			);
+		if (typeof inOptions.edge !== "boolean")
+			throw this.sequence._customError(
+				this,
+				"lockLocation",
+				"inOptions.edge must be of type boolean",
+			);
+		this._config["lockLocation"] = {
+			lock: true,
+			limit: false,
+			obj: inLocation?.object ?? inLocation,
+			offsetDistance: inOptions.offsetDistance,
+			minDistance: inOptions.minDistance,
+			maxDistance: inOptions.maxDistance,
+			edge: inOptions.edge
+		};
+		return this;
+	}
+
+	limitLocation(inLocation, inOptions = {}) {
+		inOptions = foundry.utils.mergeObject(
+			{
+				minDistance: null,
+				maxDistance: null
+			},
+			inOptions,
+		);
+		if (typeof inOptions.minDistance !== "number")
+			throw this.sequence._customError(
+				this,
+				"limitLocation",
+				"inOptions.minDistance must be of type number",
+			);
+		if (typeof inOptions.maxDistance !== "number")
+			throw this.sequence._customError(
+				this,
+				"limitLocation",
+				"inOptions.maxDistance must be of type number",
+			);
+		this._config["location"] = {
+			lock: false,
+			limit: true,
+			obj: inLocation?.object ?? inLocation,
+			minDistance: inOptions.minDistance,
+			maxDistance: inOptions.maxDistance
+		};
+		return this;
+	}
+
+	persist(inBool = true) {
+		if (typeof inBool !== "boolean")
+			throw this.sequence._customError(
+				this,
+				"persist",
+				"inBool must be of type number",
+			);
+		this._persist = inBool;
+		return this;
+	}
+
+	async run() {
+
+		const reticle = new Crosshairs({
+			t: this._type,
+			distance: this._distance,
+			direction: this._direction,
+			angle: this._angle,
+			borderColor: this._borderColor,
+			fillColor: this._fillColor,
+		}, {
+			parent: canvas.scene
+		}, this._config);
+
+		return reticle.show().then(async () => {

```

Comment:
Nitpick; would this look better to you?

```suggestion
try {
    await reticle.show()
    const position = reticle.getOrientation();
    [...]
} catch (e) {
    this.sequence._abort();
}
```

You can just ignore this if you want, I just was a little puzzled what was going on for a second when I read this.